### ISSUE TITLE
Changes to Help Bubble for more useful data

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe-old#706]
As discussed (@tiblu @ilmartyrk, 06.07), it was agreed that changing the Help Bubble to make sure it gives us more valuable data (to understand where the users main pain points are, how they are feeling when using the platform, etc,) would be a good idea, by including the following things: 

- What are you trying to do? (With space for reply)
- What difficulties are you facing? (With space for reply)
- How are you feeling? With emojis from happy to frustrated. 
- It should be possible to send comments anonymously. So instead of a mandatory email: "Want a response? Leave your email"

We discussed that this would guide the user to give us much more useful info, but that they could be stuck if they just wanted to give feedback, ask a general question, rather than having a user issue. My hunch is that they would just use one of the comment boxes anyway, but @kevincrepin (and anyone else) your thoughts on this and the idea in general would be very helpful.

**Comment by loorm** [comment]
Triage 37. Temporarily blocked, product owner need better arguments. Currently, having all this extra information looks bloated. Need to talk this over.

**Comment by BeccaMelhuish** [comment]
@loorm sorry for the slow response on this. Maybe you've already discussed it with the others by now, but in case not - here are the arguments :) (Mostly based on advice from my brother Jonathan, who is a pretty experienced UX researcher and kindly gave me a free consultation!) 

The main idea is to make the Help Centre project (including the Help request system) a way of capturing valuable data which can then be used either in the short term to make "quick fixes" to the existing platform, and/or to inform the development of Citizen OS 2.0, when bigger overhauls to the UX would be possible. The data should also help to prioritise which UX development tasks are more important than others, and therefore help to decide which things to do first (thus making the most effective use of Ilmar or Mikk's time). And the ultimate goal is that the data given is valuable enough that some day the Help Centre is no longer necessary, as the UX is so smooth and intuitive that users no longer have a need for it! 

I'm including a screenshot below of my notes from the consultation with Jonathan re the data collection side of things, which explain it a bit more thoroughly. But the main reasons for each of the additional/adjusted elements in the Help Request submission fields are:

-  **"What are you trying to do?" and "What difficulties are you having?".** This nudges the user to give us much more helpful information (in a similar way to with Github issues - "what is the problem? and "why is this a problem?". ). The more info the user gives about what they're trying to do, how they're expecting to be able to do it, what in particular is difficult/confusing/not apparent for them, the more useful the data will be for future UX improvements. 
-  **"How are you feeling?".** As UX is all about creating a smooth and positive experience for the user, knowing how they are feeling is a very important bit of feedback for us. If people enjoy using a product they're more likely to keep using it, if it causes them any frustration/anxiety (which even minor UX issues can do!) then they're less likely to. This data would especially help with prioritisation - if we are getting a lot of angry emojis selected in one particular area of the platform, we'll know to focus our efforts on improving that bit as much as possible, for example. 
- **The anonymous email field.** The email field is obviously already necessary, but making it optional to include means that users can give anonymous feedback if they want to - meaning that a frustrated user who just wants to express that they are annoyed with something that doesn't work as they expected it to, and has given up on using it so doesn't want a response, can do so freely. (Hopefully never the case, but we can't know without asking/observing!)
  
Maybe @kevincrepin has some thoughts to add here too, but I understand all the above is very good practice for UX research and therefore definitely a good move for us in my opinion :)  

![image](https://user-images.githubusercontent.com/61320526/119811394-f48f9600-bede-11eb-92d3-1625f9b1b7aa.png)

**Comment by loorm** [comment]
Thank you for providing the reasons, @BeccaMelhuish. I'm very much in agreement with the overall purpose that you stated -  that some day the Help Centre is no longer necessary, as the UX is so smooth and intuitive that users no longer have a need for it!

I also agree, that nudging users to give us more helpful information is better, than simply stating "Send us a message" as the Help Bubble currenly does. I'm not convinced, that we need the nudging to be done in two separate questions. "How can we help you?" or "What is the problem you need help with?" seem better, than "Send us a message". But we can deduce, what they were trying to do from the description of the problem they hit. If they write "Where is the bloody send to vote button?" we can safely assume, that they were trying to send the topic to vote, but could not find the button (actual user case, btw). Users, who are in trouble, do not usually have the patience to calmly and at length answer questions. We need to make it quick and easy. So what do you say we go with one field, but change the wording of the question to a more nudging one?

Asking for e-mail at the bottom, not at the top and making it optional (thereby allowing anonymous questions) both seem good and we should do it. I would, however, not add "...if you want a response" to the e-mail field, because this creates a duty for us to respond. In many cases, we would naturally respond anyway, but I wouldn't automatically make it an obligation for us.

What is the "extra data" checkbox at the very bottom?

I'm opposed to the "How are you feeling?" part, because people only click the Help button, when they're in trouble. And people who are lost or in trouble usually are not happy campers. I totally agree, that if people enjoy using something, they will keep using it. However, people who enjoy using something are not in the habit of clicking Help buttons and letting us know, that they're feeling particularly happy at the moment. What we would get from this field, then, would be a distorted picture of negativity, which gives us nothing valuable in terms of prioritisation.

In conclusion, I would be happy to push this into dev if we can kill the happy field and bring the problem entry from two fields to one. WDYT?

**Comment by kevincrepin** [comment]
_Summary of the discussion that happened on Slack._

### Current design suggestion

So I worked bit more on the help center idea which I will outline below.

The help bubble has a double purpose. It holds what it currently holds, but it can also hold the context help. I would also suggest to expand it and make it as a sidecolumn that slides in from the right. This way it can hold the content, it can be scrollable, and it doesn’t overlap the dialog etc. On mobile we don’t have much choice, here it would overlap the existing dialog…

I added a 3-level structure:
level 0 / Starting point where you can choose the general topic which you need help with or make a smart search: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=93%3A18412

Level 1 / In the general topic we can explain what is this topic and maybe show a fragment of the demo video dealing with said topic. Here we also offer more specific chapters within this topic: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=93%3A18366

Level 2 / Topic chapter – this is where the specific help bubbles would also lead to but you can also get there by the regular “navigating deeper” method. Here we can also show images or screenshots if they’re useful and we can add this question “was this useful” to gather more intel: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=93%3A18472

Finally the Submit help request is always accessible via the footer which is present on every level: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=99%3A112


### Implementation

- The content for the Help Center would live on the WEB, but be included in the APP as a widget. This way it only needs updating in one place.
- We should also have a way to link straight to the specific Help Center topic, so probably per topic we need a share URL button/link.


### Open questions

- Should FAQ and Submit a Help Request be more visible?
- Do we separate FAQ and Help Center content on a more "semantic" level so that they offer answers to very different questions? One relating to how one can interact with the platform and the other on why they should interact with it?
- The thoughts Margo raises above.

**Comment by kevincrepin** [comment]
Couple more updates.

I understand and agree with Margo's points. We can still use the feedback smileys for some other aspect, ie. after going through a flow we can ask the users to give a quick feedback for us to learn from – if we want to implement this.

So I updated the Help Request flow so that it's simpler and the entire flow is worked out now.
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=170%3A17701

I've also added icons for the categories: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=93%3A18412

And I worked out an example how it would look like if the FAQ and Help Request are more prominent (than if they're links): https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=170%3A17821
Personally I prefer them more as links, but open to comments :)

Finally I included the search in the bottom in the footer as well to be in line with how it is in the web.

-----

About the implementation on the website. @ilmartyrk can you let me know if styles should be the same or can they differ between the web and the app widget? Because for example the button style on website is different from the app, so they feel a bit out of place. 

Otherwise I guess it should be very similar? I went with that direction at least for now. Comments welcome.

Web Help Center / Level 0: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=117%3A1463
Web Help Center / Level 1: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=118%3A1546
 Web Help Center / Level 2: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=118%3A1865

Web Help Center / Request Help: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=121%3A3
Web Help Center / Request Help - Submitted: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=121%3A169

**Comment by ilmartyrk** [comment]
@kevincrepin I think styles can differ, we can include app specific stylesheet into widget if needed

**Comment by BeccaMelhuish** [comment]
@loorm @kevincrepin sorry for the delayed response again! Got a big project on at the moment. 

But yes all that Margo said sounds good to me, happy with the compromise :D And the designs Kevin has drawn up look great to me. 

A few thoughts re the outstanding questions/other points:

- I love Kevin's idea of asking for a quick feedback question with emojis after users have gone through a flow. That would indeed seems a better way to get a more unbiased overview of users emotions. 

- I also wonder if it would be worth adding in a "Give us feedback" option here (below), which would be similar to the Help request but for giving feedback (positive or negative). I.e. when someone has found the answer re how to do something, but wants to feedback that it doesn't work as they expected/there's a bug/the explanation is incorrect etc. Or that they particularly like a certain feature, or would particularly like to see a new one. Could be really useful info that comes through.    

![image](https://user-images.githubusercontent.com/61320526/121341113-4214f580-c918-11eb-8821-5b0d45c71d02.png)

- Yes perhaps we don't so much need that FAQs in their current form, once the full Help Centre is more fully up and running.. It's more useful in the interim. Though perhaps some "Quick links" (tailored to the particular page the user is on if possible @ilmartyrk ) could be useful, if there are questions which many users are navigating to much more than others. But we could track that over time. We could have a think about what would go in them otherwise, if they changed to more "abstract" questions - but "How is COS funded?" would definitely be one. 

- Re data, @ilmartyrk what kind of data would it be possible to collect from this Help Bar? Will it be possible to see which sections/questions get navigated to most often, and which also what part of the platform people are on when they navigate to certain questions? 

- How about a non-obtrusive little pop-up sliding out from the help button bottom right when people first arrive saying "We're here to help! Browse our Help Centre, submit a Help Request or give us feedback here."? So that they are aware it's there, and what they can do with it, and are encouraged to give feedback. 

- I do feel the search could be more visible on the home page, as search is a very quick and easy way to find what you're looking for. Especially if options come up as you type, but it would have to be set up well. 

- And finally what are your thoughts on testing this design @kevincrepin? My involvement with the FAQ project started with the idea that I'd help coordinate some user testing both for the FAQ "qucik fixes" and the Help Centre itself, though we've not done any yet. If you have any advice on the testing side of things, and at what point in the process we should first test the idea, to test that users will be able to easily solve their problems with this design? As I see it there are two things - the design of the Help Bar (how easy it is to navigate) but also the content we'll be putting into (how logically structured, how easy to understand). So the question is whether we test the design first (and with just mockups or after being developed?) or wait until all the content is in and test everything together. Would also be up for a quick call to discuss if easier! 

Tagging @anettlinno and @liinaguiter so they can see the discussion too now it's moved here :)

**Comment by BeccaMelhuish** [comment]
And one more thought! (Addressed to everyone).

Just looking through the FAQs project "quick fix" ideas (to try and solve the main issues as easily as possible), and lots of them involved a Help icon. 

So to be able to solve these FAQ problems as quickly as possible (i.e. without waiting until the whole Help Centre bar has been finalised and developed and all the content created etc.) shall we go ahead and create the Level 2 "specific topic" page of the Help Centre bar already, just so that there is something for the Help icons to link to right away. Then the rest of it could follow later?   

Once developed and the specific Help texts are in place we could then start one batch of testing for all the FAQs that we've actioned something to solve so far - i.e. all the ones that involve a Help icon, plus the new login process that Mikk developed (will review the FAQs and see if there are any others too). (Again, any tips re testing welcome @kevincrepin!)

**Comment by loorm** [comment]
Triage 39. Needs further discussion. @loorm will alert discussion participants in Slack #product. Moving back to "in preparation" for now.

**Comment by BeccaMelhuish** [comment]
Copying this from Slack so it's here too :) 

Hi all! Little update on this. Will post it on the Github issue too, but am posting it here as it went a little quiet over there :) 

@kevincrepin and I just had a little call about the Help Centre project, and he's blocked about a bit of time to go back to the Github issue next week, so we can hopefully make the final decisions about how the design should look. So please do chip in with your own thoughts on outstanding topics there too if you have a bit of time for it. 

We agreed we'd collaborate on doing a bit of user testing on the UI of the Help Centre (not the content yet, just the UI/navigation flow) - I'll come up with some user scenarios and testing plans, and he'll help put together the clickable/animated prototypes for it. 

We also discussed that going ahead with developing the "Level 2" part of the Help Centre already would be a good idea (it will need a slightly different design to what we have already from Kevin, just because it won't be part of a bigger flow at this stage, but pretty much as it currently is) - so that the Guidance/Help buttons can start to be put into practice already, and get some of the FAQ actions sorted (and then test that they are helping solve the problems - perhaps at the same time as we test the Help Centre UI, all in one session).   

No need to reply here though, we can continue on Github :)

@anettlinno @loorm @ilmartyrk @liinaguiter

**Comment by kevincrepin** [comment]
Updated the designs & flows a bit.

On the homepage the search is above the categories and it has been worked out how it would look if someone started typing: it would search and within the existing categories it would autocomplete the relevant ones.

I also added the additional "give us feedback" idea and made an alternative of the level 2 which could be implemented without the entire help center in place: Just ability to close the panel and fewer additional help options should do the trick.

https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=16%3A16559

**Comment by BeccaMelhuish** [comment]
Thanks @kevincrepin, this all looks great! 

Though looking at the temporary Level 2 design am now wondering how this panel coexists with the current Help Bubble (bottom right corner question mark) during the "temporary solution" phase. Is that still visible when it's open (I assume not)? 

It would make sense to me to add the "Submit a Help Request" link back to the bottom of the Level 2 panel, so that they can ask for help if they are still stuck. In which case two options would be either:

- clicking "Submit a Help Request" closes the help panel and opens the help request bubble
- we get rid of the help request bubble and go ahead with the new design for the "Submit a Help Request" bit too. In which case the "temporary solution" phase would have 2 parts - the guidance/help button "Level 2" page, and the Help request page. (In which case it might as well have the "Give us feedback" page too I think). 

I think the second option above would make more sense? Have added a "Temporary solution Becca alternative" bottom right of the Figma board which shows the slightly adjusted design for this, see if you agree? 

https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=16%3A16559

And one further question - am wondering how this bit works, as it what it looks like before (presumably both buttons are grey) and after - i.e. does the user get some kind of feedback that their feedback has been submitted? (Similar to the "help requested received" one I imagine, though I had trouble editing that to make it work myself - still getting to grips with Figma). 
 
![image](https://user-images.githubusercontent.com/61320526/122913631-9759f980-d351-11eb-9b58-555b4a8f9256.png)

**Comment by BeccaMelhuish** [comment]
Also a question more for @anettlinno 

Re the bit highlighted yellow, I feel like "and we'll look into it" is a bit non-committal for a Help Request, and if we definitely intend to get back to them (which I understand we do?) then we should say "and we'll get back to you". 

And re the bit highlighted red, as it's a help request rather than feedback, I feel like the "with your help we can make Citizen OS better" doesn't work so well, as in the user's mind they are asking us for help rather than helping us. Perhaps a response time commitment, e.g. "We successfully received your request! We aim to respond to all requests within X days" could go their instead?

![image](https://user-images.githubusercontent.com/61320526/122916407-c160eb00-d354-11eb-8139-7e12ddcafe85.png)

**Comment by anettlinno** [comment]
@BeccaMelhuish @kevincrepin all looks really great! Only one question -- according to @ilmartyrk the Help Center will be hosted on our web site from where the Help Bubble will get the information. We would need design for the web as well but I cannot find if from the Figma. Do we have it somewhere separately or do we need to create this one too? (sorry if I´ve missed it from the feed)

@BeccaMelhuish regarding the "Submit a help request" then your suggestion would work well I think:
"_If you didn´t find answer from our Help Center, please write us and we´ll get back to you_". We have internal agreement that we´ll write back in 24 hours on work days but this would go too long... 

AND as for the text in blue box I agree that in this point users are rather looking for help and "With your help... " is not quite what they´d want to read here, but I don´t have any better idea for the text below the main message now :)

**Comment by kevincrepin** [comment]
Hey @anettlinno, the website version is visible here: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=117%3A1463

@BeccaMelhuish good points! I added the flow  of feedback when submitting the "was this helpful" question. 

About the help bubble as it exists...I think we can actually still keep it? If you click on it, it opens the help request. If it's open and you click on a context help bubble, it will just open that one and close the help request. Because otherwise the only way people can reach the help request is by finding a context bubble somewhere, which they would need to go and find.

I added these temporary flows here: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=276%3A241

**Comment by anettlinno** [comment]
Thanks a lot for the web site design link, @kevincrepin!

I think that if user clicks on the Help Bubble then Level 0 should open and not the existing one:

<img width="185" alt="Help Bubble" src="https://user-images.githubusercontent.com/45175679/123610747-8d307300-d809-11eb-870d-d5051174e434.png">

**Comment by BeccaMelhuish** [comment]
Hi all! Am just clocking off on my holidays ('til the 6th) but have had a quick skim read of these last messages, will have a more thorough look when I'm back but on the whole all sounds good to me :) 

I can suggest some texts when I'm back if needed. Ciao for now!

**Comment by liinaguiter** [comment]
Hi everyone! 

Regarding the context sensitive help icon, it has been agreed now, that they will be created. Correct me if I am wrong.
I am talking about the question mark icon that will be used in places that create frequent problems. 
![Screenshot 2021-09-16 at 00 34 08](https://user-images.githubusercontent.com/34899902/133512956-18e73ac4-3a8d-4123-a8fb-0b4cd27dcffd.png) 
@kevincrepin has this icon beed designed for the platform? 
@ilmartyrk, does this need to be a separate issue or can it be considered as part of the Help center programming?

**Comment by liinaguiter** [comment]
Hi all!

**Help Center update:**

Please all add to / correct this summary, if something is missing. I'll arrange a meeting for everyone to attend together, to go over all deadlines, understand who does what and to keep the momentum going. Work package and timeline: https://tinyurl.com/3s5yryy5

Sepp + Argo met. Ilmar + Argo met. Kevin and Becca have been tweaking the UX design in Figma. Becca is working on the content/ texts. Becca has gotten at least five potential testers via a social media campaign. 

**The next steps**  
- Becca is getting the testing scenarios ready today, Kevin will work out how to test them. They need some time to discuss and ping-pong ideas. 
- Then we are ready for testing, (test design only).
- Then possibly make some tweaks depending on the results .
- Then Argo can start to program the website version. Becca will keep on writing texts at the same time. 

**Meeting notes in general:**
- All texts live on the website. The Help Center widget is on the platform.
- Search function to work with different wordings - how to join a group, joining a group, group joining.
- The 'interim version' is not being developed. But context sensitive help icon (question mark) will be developed and added to places where people get stuck frequently (relevant current FAQ).
- FAQ will stay, but change - it will be basically a condensed version of the website. 
- Help Center is a user manual. 
- Existing 'Help topics' on the platform will be deleted.   **Meeting notes and updates on specific areas:

**Designs**
These are most up-to-date versions of the designs: 
- Website version: https://www.figma.com/file/eHBSunqUv10fMcWmiEEXiv/Website?node-id=117%3A1463
- Platform version: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=16%3A16559

**Translation:**
- English first, then Estonian only. Liina has agreed to translate smaller batches and review the translations that will be made by the Estonian by the agency. 
- Path is translated, but URL is not translated - should be hidden in CrowdIn.

**Content:** 
- Anett has gone through the whole functionality of our platform - step-by-step and created the backbone of the structure for texts.
- Texts writing - start with the bare minimum - what we can launch. Becca writes all texts from scratch in COS style. Texts come in batches, so that we can launch the help center already with first content uploaded. The rest will continue. 
- At first will produce texts + screenshots. Then will make videos. Videos will be inserted via Youtube at first, can be changed later. 
- Testing of the texts needs to be done with users.

**Testing**
- Almost ready to start the testing of the help center design.
- Texts / content will be tested later.

**Comment by BeccaMelhuish** [comment]
All looks good to me thank you @liinaguiter!

The only thing I would clarify for the others is the FAQs. We decided in the Help Centre we'll have "Quick links" on the Help Centre home page, which will be the most frequently clicked on Help Centre topics (user issues). Then on the Citizen OS website we'll change the FAQs page to being more about Citizen OS foundation more broadly - who we are, what we do, who funds us, etc. (i.e. no longer anything about user issues on the platform).

**Comment by seppviljar** [comment]
Re: translations:
So, the guidance widget will always be in English (except when the app is in Estonian) or the widget framework is translated but topics are not?

BTW, if the reason to keep languages bare minimum is cost then there's also possibility that
- volunteers want to translate help section for free, for the app
- machine translation is "good enough" to replace English version

So I'd recommend setting up the backend in a way that, in the future, if we suddenly want to translate HelpCenter into a language that's in app but not in the web (a la Polish) then we could do it without significant overhaul.

**Comment by anettlinno** [comment]
Thanks @liinaguiter for the overview! Almost there :) 

Regarding the translations then Agustina and Indo team + volunteers are going to translate the Help Center in one point, but it doesn´t affect launching it in English + Estonia translation. @seppviljar good idea to set up the backend so that it would be possible to translate the Help Center in app language if there´s a need for that.

**Comment by liinaguiter** [comment]
Argo will develop now, made a new issue only for the Help Center, can close this issue.

**Comment by ilmartyrk** [comment]
move to https://github.com/citizenos/citizenos-fe/issues/56
